components/job-orchestration/job_orchestration/executor/query/extract_stream_task.py

```
@@ -33,6 +33,19 @@
 # Setup logging
 logger = get_task_logger(__name__)
 
+# Module-level cached SqlAdapter to reuse DB connections across tasks within the same worker.
+_cached_sql_adapter: SqlAdapter | None = None
+_cached_sql_adapter_params: dict | None = None
+
+
+def _get_or_create_sql_adapter(clp_metadata_db_conn_params: dict) -> SqlAdapter:
+    global _cached_sql_adapter, _cached_sql_adapter_params
```

Ruff (0.15.7)

[warning] Using the global statement to update `_cached_sql_adapter` is discouraged

(PLW0603)